morpheus/_lib/src/stages/kafka_source.cpp

```diff
@@ -351,13 +358,37 @@ KafkaSourceStage::KafkaSourceStage(const neo::Segment &parent,
                 return m_requires_commit;
             });
 
+        auto &context = srf::runnable::Context::get_runtime_context();
+
         // Build consumer
-        m_rebalancer  = &rebalancer;
-        auto consumer = this->create_consumer();
+        // m_rebalancer  = &rebalancer;
+        auto consumer = this->create_consumer(rebalancer);
```

Comment:
Dual of the .hpp file comment, if the rebalancer needs to be created externally, it should moved to its own file and have the `KafkaSourceStage__` prefix removed.

Reply:
Done